common/namespace/registry.go

```diff
@@ -227,6 +228,17 @@ func (r *registry) GetCacheSize() (sizeOfCacheByName int64, sizeOfCacheByID int6
 	return int64(r.cacheByID.Size()), int64(r.cacheNameToID.Size())
 }
 
+func (r *registry) InvalidateCacheKey(name Name, id ID) {
+	r.cacheLock.Lock()
+	r.cacheByID.Delete(id)
+	r.cacheLock.Unlock()
+
+	r.readthroughLock.Lock()
```

Comment:
this should before cacheLock

Reply:
I am thinking it is too risky to acquire 2 locks. I changed the order of lock. Let me know if you have any further concern.

Reply:
That's fine, if you look at getOrReadthroughNamespaceByID impl, it first acquires readthroughLock and then the cacheLock.

But why not just call `getNamespaceByIDPersistence` and then `updateCachesSingleNamespace` ? Then we don't need to remove anything from cache.